### PR TITLE
Add Vec3.generateOrthogonal

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

* Add `Vec3.generateOrthogonal` API to generate an orthogonal vector for specified vector. This method was once used internally and now public to use.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by stanleyljl — APPROVED** [review]

File: cocos/core/math/vec3.ts [hunk 1]
```diff
@@ -801,6 +801,32 @@ export class Vec3 extends ValueType {
...
 
+    /**
+     * @zh 生成指定向量的一个正交单位向量。如果指定的向量 **精确地** 是零向量，则返回 **精确的** 零向量。
+     * @en Generates an unit vector orthogonal to specified vector.
+     * If the specified vector is **strictly** zero vector, the result is **strict** zero vector.
+     * @param out @zh 生成的向量。@en The generated vector.
+     * @param n @zh 输入向量。该向量 **不必** 是标准化的。 @en The input vector. **Need not** to be normalized.
+     * @returns `out`
+     */
+    public static generateOrthogonal<Out extends IVec3Like> (out: Out, n: Readonly<IVec3Like>): Out {
```
Comment (minggo) [inline cocos/core/math/vec3.ts:812]: It is better not use IVec3Like.

Comment (minggo) [reply cocos/core/math/vec3.ts:812]: Just use Vec3

Comment (shrinktofit) [reply cocos/core/math/vec3.ts:812]: Why? I just following the existing style

Comment (minggo) [reply cocos/core/math/vec3.ts:812]: Because IVec3Like is duck type. And the original implementation also use Vec3.

Comment (minggo) [reply cocos/core/math/vec3.ts:812]: > Why? I just following the existing style

After discussion, i think it is reasonable. We should modify all the functions at the same time.

**Review by minggo — APPROVED** [review]

**Pull request merged**